### PR TITLE
[THORN-2160] Including JWE key encryption alias as a header by default

### PR DESCRIPTION
**Pull request by sberyozkin**

### CODE REVIEW

**Comment by sberyozkin** [comment]
Jose4J examples page says it is a 'polite thing to do' to report a (public) JWE encryption key alias :-), hence, as we agreed with David, that should be done by default

**Comment by sberyozkin** [comment]
Michal or Ladislav, please have a quick look, these are just minor updates across two tested implementations (the default one and Jose4j)

File: fractions/jose/src/main/java/org/wildfly/swarm/jose/JoseConfiguration.java [hunk 1]
```diff
@@ -257,4 +267,11 @@ public String contentEncryptionAlgorithm() {
     @Configurable("thorntail.jose.encryption.key.alias")
     @AttributeDocumentation("Alias to the encryption key entry in the keystore")
     private Defaultable<String> encryptionKeyAlias = string(DEFAULT_KEY_ALIAS);
+
+    /**
+     * Include Encryption Key Alias as Header.
+     */
+    @Configurable("thorntail.jose.encryption.include.alias")
+    @AttributeDocumentation("Include the encryption key alias as a 'kid' header: true (default)")
```
Comment (Ladicek) [inline fractions/jose/src/main/java/org/wildfly/swarm/jose/JoseConfiguration.java]: What "header"? It's not an HTTP header, right?

Also, I'd replace `: true (default)` with ` (defaults to true)`.

**Comment by Ladicek** [comment]
I don't quite understand what "header" are we talking about, but if you clarify that in the attribute documentation, I think that would be enough.

Otherwise LGTM.

File: fractions/jose/src/main/java/org/wildfly/swarm/jose/JoseConfiguration.java (hunk 1 above)
Comment (dvddhln) [reply fractions/jose/src/main/java/org/wildfly/swarm/jose/JoseConfiguration.java]: It is the JWE(JSON Web Encryption) header it is referring to. Described in detail here
https://tools.ietf.org/html/rfc7516

Comment (Ladicek) [reply fractions/jose/src/main/java/org/wildfly/swarm/jose/JoseConfiguration.java]: That RFC uses a term "JOSE header", so if this read "Include the encryption key alias as the JOSE `kid` header", or something like that, that would be great IMHO.

Comment (sberyozkin) [reply fractions/jose/src/main/java/org/wildfly/swarm/jose/JoseConfiguration.java]: Sure, np, will update shortly

Comment (dvddhln) [reply fractions/jose/src/main/java/org/wildfly/swarm/jose/JoseConfiguration.java]: Sounds good to me!

**Review by Ladicek — APPROVED** [review]

**Comment by sberyozkin** [comment]
retest this please

**Comment by sberyozkin** [comment]
Lets see what happens to https://ci.wildfly-swarm.io/job/thorntail-pull-request-linux/717/ :-)

**Comment by sberyozkin** [comment]
Jose tests are green

**Pull request merged**